### PR TITLE
Removes JaxbLink and related classes and tests

### PR DESCRIPTION
**Pull request by spericas**
Removes JaxbLink and related classes and tests. Includes some minor cleanup of versions and version management in pom files too. Issue #1226.

### CODE REVIEW

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java [hunk 1]
```diff
@@ -16,18 +16,11 @@
```
Comment (jim-krueger) [inline jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:16]: Change to 2024

File: jaxrs-api/src/main/java/module-info.java [hunk 2]
```diff
@@ -19,8 +19,6 @@
  */
 module jakarta.ws.rs {
```
Comment (jim-krueger) [inline jaxrs-api/src/main/java/module-info.java:20]: Change to 2024

**Review by jim-krueger — COMMENTED** [review]
Just a couple copyright changes, sorry.

**Review by jim-krueger — APPROVED** [review]
Looks good!

**Review by jamezp — APPROVED** [review]
This LGTM. The only other thing that needs to be done is the signature test file needs to be updated. That said, it looks like the signature test file is not correct as I'm seeing it fail locally for other reasons. We could just update it in a follow up PR.

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (jim-krueger) [inline jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:2]: Sorry,  according to https://www.eclipse.org/projects/handbook/#ip-copyright-headers this should really be just
`Copyright (c) 2011`
The same with the rest.
Sorry for the confusion!

**Review by jim-krueger — COMMENTED** [review]
Sorry for the copyright confusion.   Shame on me for assuming that copyright practices would be universal. https://www.eclipse.org/projects/handbook/#legaldoc-faq

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java (hunk 3 above)
Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:2]: Now is it wrong to have two years (create and modify) or is it simply no longer required? There's many files with two-year copyrights we probably don't want to update at this time.

Comment (jim-krueger) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:2]: Apparently it is wrong to have two years  [](https://www.eclipse.org/projects/handbook/#legaldoc-faq) .  And yes, I see our code is all of the board in this respect.   I certainly wouldn't be in favor of fixing every file and I probably wouldn't have even brought it up if I hadn't requested that you make an incorrect change earlier (sorry).   @mkarg brought this up in a review of one of my PRs that was also incorrect.   And I will try to follow the rules going forward and I'll probably fix any part I touch that is incorrect.

Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:2]: Ok, I just updated those two files again

**Review by jamezp — APPROVED** [review]

**Review by jim-krueger — APPROVED** [review]

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java (hunk 3 above)
Comment (mkarg) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/Link.java:2]: > Now is it wrong to have two years (create and modify) or is it simply no longer required? There's many files with two-year copyrights we probably don't want to update at this time.

According to the Eclipse Handbook it is not necessarily wrong or forbidden to have two years, but having the second year is not adviced anymore. Hence to not make it wrong, it is easiest to remove the second date whenever we touch a file, but there is no need to remove the second date unless the file is touched anyways.

**Review by jansupol — APPROVED** [review]

**Pull request merged**